lib/ast-utils.js

```diff
@@ -247,20 +247,213 @@ function isParenthesised(sourceCode, node) {
 }
 
 /**
- * Gets the `=>` token of the given arrow function node.
+ * Checks if the given token is an arrow token or not.
  *
- * @param {ASTNode} node - The arrow function node to get.
- * @param {SourceCode} sourceCode - The source code object to get tokens.
- * @returns {Token} `=>` token.
+ * @param {Token} token - The token to be check.
+ * @returns {boolean} `true` if the token is an arrow token.
  */
-function getArrowToken(node, sourceCode) {
-    let token = sourceCode.getTokenBefore(node.body);
+function isArrowToken(token) {
+    return token.value === "=>" && token.type === "Punctuator";
+}
 
-    while (token.value !== "=>") {
-        token = sourceCode.getTokenBefore(token);
-    }
+/**
+ * Checks if the given token is a comma token or not.
+ *
+ * @param {Token} token - The token to be check.
+ * @returns {boolean} `true` if the token is a comma token.
+ */
+function isCommaToken(token) {
+    return token.value === "," && token.type === "Punctuator";
+}
+
+/**
+ * Checks if the given token is a comma token or not.
+ *
+ * @param {Token} token - The token to be check.
+ * @returns {boolean} `false` if the token is a comma token.
+ */
+function isNotCommaToken(token) {
```

Comment:
Another possibility would be maintaining the current API but using the helper to pre-negate the positive definitions.

```js
function isCommaToken(token) {
    return token.value === "," && token.type === "Punctuator";
}

const isNotCommaToken = negate(isCommaToken);
```

With either solution, we get to avoid maintaining a set of implementations as well as their inverses separately.